homeassistant/helpers/script.py

```diff
@@ -142,8 +142,24 @@ def async_script_wait(entity_id, from_s, to_s):
                 if self._change_listener:
                     self.hass.async_add_job(self._change_listener)
 
+                # Allow proceed to be false by default
+                proceed = False
+
+                if CONF_PROCEED in action:
+                    proceed = action[CONF_PROCEED]
+
                 if CONF_TIMEOUT in action:
-                    self._async_set_timeout(action, variables)
+                    # Check if we want to proceed to execute
+                    # the script after the timeout
+                    # or exit as the default behaviour
+                    if proceed:
```

Comment:
Actually, if I were you, I will extend `self._async_set_timeout` method to accept the 3rd parameter as the callback method when timeout occurred. If the 3rd parameter is None, then keep current logic. If we want to continue run after timeout, we can simply pass in `async_script_wait`.